src/frontend/src/common/components/library/data_subview/index.tsx

```diff
@@ -102,6 +129,115 @@ const DataSubview: FunctionComponent<Props> = ({
     searching: false,
   });
 
+  const [checkedSamples, setCheckedSamples] = useState<any[]>([]);
+  const [isHeaderChecked, setIsHeaderChecked] = useState<boolean>(false);
+  const [showCheckboxes, setShowCheckboxes] = useState<boolean>(false);
+  const [isHeaderIndeterminant, setHeaderIndeterminant] =
+    useState<boolean>(false);
+  const [open, setOpen] = useState(false);
+  const [isDownloadDisabled, setDownloadDisabled] = useState<boolean>(true);
+  const [failedSamples, setFailedSamples] = useState<string[]>([]);
+  const [downloadFailed, setDownloadFailed] = useState<boolean>(false);
+  const [isMetadataSelected, setMetadataSelected] = useState<boolean>(false);
+  const [isFastaSelected, setFastaSelected] = useState<boolean>(false);
+  const [isFastaDisabled, setFastaDisabled] = useState<boolean>(false);
+
+  const handleDownloadClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleDownloadClose = () => {
+    setOpen(false);
+    setMetadataSelected(false);
+    setFastaSelected(false);
+  };
+
+  useEffect(() => {
+    // add all samples if header checkbox is selected
+    if (isHeaderChecked) {
+      const allPublicIds: any[] = [];
```

Comment:
`allPublicIds` should be just `string[]`?

is it difficult to type `failedSamples`? Just trying to see if we can avoid `any` as much as possible 😆

Reply:
i'll switch it back! I think there was some strange type stuff going on with it at the time, but i'll try adding back string[]!